File: leetcode/java/non-overlapping-intervals.java
import java.util.*;
class Solution{
    public void eraseOverlapIntervals(int[][] intervals){
        Arrays.sort(intervals, (a,b)->Integer.compare(a[1], b[1]));
        System.out.println(intervals);
    }
    public static void main(String[] args){
        Scanner scanner = new Scanner(System.in);
        Solution A = new Solution();
        System.out.println("Print the length of the array");
        int length = scanner.nextInt();
        List<List<Integer>> numbers = new ArrayList<Integer>();
        for(int i = 0; i < length; i++){
            int tempOne = scanner.nextInt();
            int tempTwo = scanner.nextInt();
            var newArray = new ArrayList<Integer>();
            newArray.add(tempOne);
            newArray.add(tempTwo);
            numbers.add(newArray);
        }
        scanner.close();
    }
}

File: leetcode/python/non-overlapping-intervals.py
from typing import List
class Solution:
    def eraseOverlapIntervals(self, intervals: List[List[int]]) -> int:
        intervals.sort(key=lambda x: x[1])
        print(intervals)
        end,count = intervals[0][1],1
        for i in range(1,len(intervals)):
            if intervals[i][0] >= end:
                count+=1
                end = intervals[i][1]
        return len(intervals) - count